pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -445,7 +445,7 @@ func (r *components) handleDataInput(ctx context.Context, ins []v1alpha2.DataInp
 		}
 		if !ready {
 			uds = append(uds, makeUnsatisfiedDependency(obj, s, in, reason))
-			return uds, nil
+			continue
```

Comment:
It makes sense when dependency condition unsatisfied, can you give a case to show why we MUST use `continue` here? Thanks.

Reply:
it didn't need to loop check other `v1alpha2.DataInput` ？

Reply:
this return with condition unsatisfied will lead to another check/reconcile soon.

Reply:
ok